### PR TITLE
task: add short-lived token support in `rhacs-*` tasks

### PR DESCRIPTION
**Pull request by rhybrillou**
# Changes

Add support for short-lived tokens in `rhacs-deployment-check`, `rhacs-image-check` and `rhacs-image-scan` tasks.

The above three tasks have a new version that can be configured either with authorization using API tokens (similar configuration to `3.71` version of the tasks), or with authorization using short-lived tokens received from the new `rhacs-m2m-authenticate` task.

Both configuration variants have a sample with the expected `Secret`, `Pipeline` and `PipelineRun` objects.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [ ] Includes [tests][tests] (for new tasks or changed functionality)
  - **No**. _Requires external service integration._
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Comment by vinamra28** [comment]
@rhybrillou, will it be possible for you to break the PR into multiple PRs ? 😅

**Comment by dhaus67** [comment]
@vinamra28 I'm working together with @rhybrillou for the series of PRs, so I took the liberty to split them:
- this PR now only contains the updates to `rhacs-deployment-check` (i.e. a new version).
- #1260 covers the new version for `rhacs-image-scan`.
- #1261 covers the new version for `rhacs-image-check`.
- #1259 covers the introduction of a new authentication task for all `rhacs-*` tasks.

Hope this split makes things easier to review!

**Review by vdemeester — APPROVED** [review]

File: task/rhacs-deployment-check/4.0/rhacs-deployment-check.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+metadata:
+  name: rhacs-deployment-check
+  labels:
+    app.kubernetes.io/version: "4.0"
+  annotations:
+    tekton.dev/tags: security
+    tekton.dev/categories: Security
+    tekton.dev/displayName: "Policy check a deployment with Red Hat Advanced Cluster Security"
+    tekton.dev/platforms: "linux/amd64,linux/arm64,linux/ppc64le,linux/s390x,darwin/amd64,darwin/arm64,windows/amd64"
+    tekton.dev/pipelines.minVersion: "0.43.0"
```
Comment (vinamra28) [inline task/rhacs-deployment-check/4.0/rhacs-deployment-check.yaml:13]: I guess `tekton.dev/v1` was introduced in `0.50.0` can we set that version here? 😅

**Comment by vinamra28** [comment]
can you please squash the commits? will merge post that

**Pull request merged**